src/excmds.ts

```diff
@@ -2212,18 +2212,35 @@ export async function tabopen(...addressarr: string[]) {
         }
     }
 
-    return activeTabContainerId().then(containerId => {
-        const args = { active } as any
-        // Ensure -c has priority.
-        if (container) {
-            if (container !== "firefox-default") {
-                args.cookieStoreId = container
+    const containerId = await activeTabContainerId()
+    const args = { active } as any
+    // Ensure -c has priority.
+    if (container) {
+        if (container !== "firefox-default") {
+            args.cookieStoreId = container
+        }
+    } else if (containerId && config.get("tabopencontaineraware") === "true") {
+        args.cookieStoreId = containerId
+    }
+    const maybeURL = await queryAndURLwrangler(query)
+    if (typeof maybeURL === "string") {
+        return openInNewTab(maybeURL, args)
+    }
+
+    if (typeof maybeURL === "object") {
+        if (await firefoxVersionAtLeast(80)) {
+            // work around #2695 until we can work out what is going on
+            if (args.active === false || args.cookieStoreId !== undefined) {
+                throw new Error("Firefox search engines do not support containers or background tabs in FF >80. `:set searchengine google` or see issue https://github.com/tridactyl/tridactyl/issues/2695")
             }
-        } else if (containerId && config.get("tabopencontaineraware") === "true") {
-            args.cookieStoreId = containerId
+
+            // This ignores :set tabopenpos / issue #342. TODO: fix that somehow.
+            return browser.search.search(maybeURL)
```

Comment:
@glacambre at the heart of it it's quite a simple change: I don't open a new tab and then loadreplace it, instead we open a new tab directly.

I don't know why your method stopped working reliably in FF80. I don't really know how I knew it was your method that was the problem either - it was just a hunch as it was the most convoluted thing I could find to do with tabopen. I thought it looked a bit janky in use so I thought it was a good thing to remove anyway :)